### PR TITLE
Improve livesync error handling

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
* Minor code improvements - use of cache decorator, removing redundant interfaces
* Error handling imrpovements - attach more information to the error object (deviceIdentifier), aggregate errors
* Enable hooks to run successfully when  CLI is `require()`-d

Ping @rosen-vladimirov @TsvetanMilanov

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: declarations.d.ts [hunk 1]
```diff
@@ -1824,3 +1824,26 @@ interface IDeferPromise<T> extends IPromiseActions<T> {
...
+
+	/**
+	 * Posts a notification.
+	 * @param {string} deviceIdentifier Device's identifier.
+	 * @param {string} notification The xml value of the Name key of the notification to be post.
+	 * @param {string} @optional commandType The xml value of the Command key of the notification to be post.
+	 * @return {Promise<number>} A socket which can be queried for a response.
+	 */
+	postNotification(deviceIdentifier: string, notification: string, commandType?: string): Promise<number>;
+}
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: new line at the end, please

File: mobile/ios/device/ios-application-manager.ts [hunk 2]
```diff
@@ -92,8 +93,13 @@ export class IOSApplicationManager extends ApplicationManagerBase {
 	}
 
 	public async restartApplication(applicationId: string, appName?: string): Promise<void> {
-		await this.stopApplication(applicationId, appName);
-		await this.runApplicationCore(applicationId);
+		try {
+			await this.stopApplication(applicationId, appName);
+			await this.runApplicationCore(applicationId);
+		} catch (err) {
+			await this.$iOSNotificationService.postNotification(this.device.deviceInfo.identifier, `${applicationId}:NativeScript.LiveSync.RestartApplication`);
```
Comment (rosen-vladimirov) [inline mobile/ios/device/ios-application-manager.ts]: so... in case we are restarting Cordova App ?
Or NativeScript app built with Telerik Platform?

Comment (Mitko-Kerezov) [inline mobile/ios/device/ios-application-manager.ts]: This is handled in `appbuilder` subfolder in `ios-livesync-service.ts`. This change is irrelevant regarding Progress Telerik Platform and Cordova, as can be seen in the [file itself](https://github.com/telerik/mobile-cli-lib/blob/master/appbuilder/services/livesync/ios-livesync-service.ts#L26).

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**